src/main/java/me/contaria/seedqueue/mixin/server/MinecraftServerMixin.java

```diff
@@ -183,9 +183,19 @@ private void pauseServer(MinecraftServer server, boolean value, Operation<Void>
         if (!entry.hasWorldPreview()) {
             return false;
         }
-        if (!entry.isLocked() && SeedQueue.config.maxWorldGenerationPercentage < 100) {
+        if (entry.isLocked()) {
+            return false;
+        }
+        if (SeedQueue.config.resumeOnFilledQueue && SeedQueue.noPrioritizedRemaining() && !SeedQueue.hasMoreCapacity()) {

```

Comment:
I'm a bit concerned about the call to `SeedQueue.noPrioritizedRemaining()` potentially causing performance issues. I've had issues before with iterating the queue concurrently in this method.
Could you switch the order of `SeedQueue.noPrioritizedRemaining()` and `!SeedQueue.hasMoreCapacity()` and also move it to only be called if an entry is about to be paused based on `SeedQueueConfig#maxWorldGenerationPercentage`?

Reply:
Hmm quick suggesh, thoughts on synchronising it with the LOCK in SeedQueue? Does that fix anything at all? Also I didn't get what you mean by only running it based on maxWorldGenPercent... Is it like... I only run the check if that config option is < 100?

Reply:
I just realized i dont really need that call to `SeedQueue.noPrioritizedRemaining()` i can just check if `SeedQueueEntry.isPrioritized()` and it'll work the same way because it is gonna check everything anyways.. nice catch!!